src/app-layer-htp.c

```diff
@@ -1947,22 +1821,26 @@ static int HTPCallbackRequestBodyData(htp_tx_data_t *d)
 
 end:
     if (hstate->conn != NULL) {
-        SCLogDebug("checking body size %"PRIu64" against inspect limit %u (cur %"PRIu64", last %"PRIu64")",
-                tx_ud->request_body.content_len_so_far,
-                hstate->cfg->request.inspect_min_size,
-                (uint64_t)hstate->conn->in_data_counter, hstate->last_request_data_stamp);
+        SCLogDebug("checking body size %" PRIu64 " against inspect limit %u (cur %" PRIu64
+                   ", last %" PRIu64 ")",
+                tx_ud->request_body.content_len_so_far, hstate->cfg->request.inspect_min_size,
+                (uint64_t)htp_conn_request_data_counter(hstate->conn),
+                hstate->last_request_data_stamp);
 
         /* if we reach the inspect_min_size we'll trigger inspection,
          * so make sure that raw stream is also inspected. Set the
          * data to be used to the amount of raw bytes we've seen to
          * get here. */
         if (tx_ud->request_body.body_inspected == 0 &&
             tx_ud->request_body.content_len_so_far >= hstate->cfg->request.inspect_min_size) {
-            if ((uint64_t)hstate->conn->in_data_counter > hstate->last_request_data_stamp &&
-                (uint64_t)hstate->conn->in_data_counter - hstate->last_request_data_stamp < (uint64_t)UINT_MAX)
-            {
-                const uint32_t data_size = (uint32_t)(
-                        (uint64_t)hstate->conn->in_data_counter - hstate->last_request_data_stamp);
+            if ((uint64_t)htp_conn_request_data_counter(hstate->conn) >
+                            hstate->last_request_data_stamp &&
+                    (uint64_t)htp_conn_request_data_counter(hstate->conn) -
+                                    hstate->last_request_data_stamp <
+                            (uint64_t)UINT_MAX) {
```

Comment:
This isn't very easy to read, as is...

Reply:
What do you propose ?

Reply:
GH didn't keep the suggestion in this thread :/

Reply:
:-/